src/consumers.ts

```diff
@@ -109,51 +133,69 @@ class EmptyResultEngine extends AnalysisConsumer implements DiagnosticProducer
-            this.item.finished_at === undefined ||
-            this.item.finished_at == null) {
+        if (this.item == {} && (this.item.finished_at === undefined ||
+            this.item.finished_at == null)) {
             return [{
                 severity: DiagnosticSeverity.Information,
                 range: get_range(this.context.version),
                 message: `Application dependency ${this.context.name.value}-${this.context.version.value} - analysis is pending`,
-                source: 'Dependency Analytics'
+                source: 'Dependency Analytics '
             }]
         } else {
             return [];
         }
     }   
 }
 
+let targer_link : DocumentUri;
+
 /* Report CVEs in found dependencies */
 class SecurityEngine extends AnalysisConsumer implements DiagnosticProducer
 {
     constructor(public context: IDependency, config: any) {
         super(config);
-        this.binding = {path: ['result', 'recommendation', 'component-analyses', 'cve']};
+        this.binding = {path: ['component_analyses', 'vulnerability']};
         /* recommendation to use a different version */
-        this.changeToBinding = {path: ['result', 'recommendation', 'change_to']};
+        this.changeToBinding = {path: ['recommended_versions']};
+        /* snyk registration link */
+        this.regLinkBinding = {path: ['registration_link']};
+        /* Diagnostic message */
+        this.messageBinding = {path: ['message']};
+        /* Public and Private vulnerability count */
+        this.pubVulBinding = {path: ['known_security_vulnerability_count']};
+        this.pvtVulBinding = {path: ['security_advisory_count']};
     }
 
     produce(ctx: any): Diagnostic[] {
         if (this.item.length > 0) {
-            let cveList = [];
-            for (let cve of this.item) {
-                cveList.push(cve['id'])
+            /* Counting total # of Public and Private Vulnerability */
+            Vul_private += this.pvtVul;
+            Vul_public += this.pubVul;
+            /* Assign a string to a type DocumentUri */
+            targer_link = this.regLink;
+            /* The diagnostic's severity. */
+            let diag_severity;
+
+            if (this.pubVul == 0 && this.pvtVul > 0) {
+                diag_severity = DiagnosticSeverity.Information; 
+            } else {
+                diag_severity = DiagnosticSeverity.Error;
             }
-            let cves = cveList.join(' ');
 
             let diagnostic = {
-                severity: DiagnosticSeverity.Error,
+                severity: diag_severity,
                 range: get_range(this.context.version),
-                message: `Application dependency ${this.context.name.value}-${this.context.version.value} is vulnerable: ${cves}`,
-                source: 'Dependency Analytics'
+                message: `${this.message}`,
+                source: 'Dependency Analytics ',
+                code: `Find out more: ${targer_link}` 
```

Comment:
You can directly assign `this.regLink` instead of referring global var

Reply:
`this.regLink` is type of string and before passing a URL to diagnostic it should be converted to a URI object of type "DocumentUri" `targer_link : DocumentUri;` for a URL.

Reply:
https://microsoft.github.io/language-server-protocol/specification#textDocuments